storage/StorageExampleSwiftUI/ImagePicker.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  ImagePicker.swift
+//  StorageExampleSwiftUI
+//
+//  Created by Ben Reed on 3/25/21.
+//  Copyright © 2021 Google Inc. All rights reserved.
+//
+
+import SwiftUI
+import Firebase
+
+struct ImagePickerRepresentable {
+    enum Source {
+        case camera
+        case photoLibrary
+    }
+    
+    var source: Source
+    @ObservedObject var store: ImageStore
+    @Binding var visible: Bool
+    var completion: () -> Void
+    
+    class Coordinator: NSObject {
+        private var representable: ImagePickerRepresentable
+        private var store: ImageStore
+        
+        init(representable: ImagePickerRepresentable, store: ImageStore) {
+            self.representable = representable
+            self.store = store
+        }
+    }
+}
+
+extension ImagePickerRepresentable.Coordinator: UIImagePickerControllerDelegate {
```

Comment:
If you haven't considered it, check out [`PHPickerViewController`](https://developer.apple.com/documentation/photokit/phpickerviewcontroller) as well. There's a cool [WWDC 2020 video](https://developer.apple.com/videos/play/wwdc2020/10652/) where I learned about it

Reply:
ive read a few cool articles about it. like this [this](https://nemecek.be/blog/30/checking-out-the-new-phpickerviewcontroller-in-ios-14-to-select-photos-or-videos) one

Reply:
Hey @ncooke3, I really wanted to use it too haha. I started with the PHPickerViewController. I got it working but I found two issues:

1. PHPickerViewController does not provide any access to the camera yet, so I was forced to also incorporate UIImagePickerController or get a camera manually.

2. There is a bug I found when using search in the PHPickerViewController and SwiftUI on an actual device. On my iPhone 12 Pro Max, the picker view process crashes whenever using the search bar. The rest of the controller behaves as usual. I opened feedback with Apple to address it.

Reply:
I didnt know that but that's good to know! 👍